test/new_delete.cpp

```diff
@@ -0,0 +1,23 @@
+#include "../../relacy/relacy_std.hpp"
+#include "../../relacy/relacy_cli.hpp"
+
+#include <vector>
+
+struct new_delete_tests : rl::test_suite<new_delete_tests, 1>
+{
+    void thread(unsigned index)
+    {
+        std::vector<int> vs;
```

Comment:
This test fails with c++14 and newer on GCC, since the allocation conditionally calls `operator delete(void*, size_t)`. Adding the overload to context.hpp addresses the crash.